### PR TITLE
chore: update ob addon

### PR DESCRIPTION
**Pull request by shanshanying**
- to #1181

### CODE REVIEW

**Review by wangyelei — APPROVED** [review]

File: addons/oceanbase-ce/dataprotection/backup.sh [hunk 1]
```diff
@@ -1,135 +1,45 @@
-set -e
+# !/bin/bash
```
Comment (Y-Rookie) [inline addons/oceanbase-ce/dataprotection/backup.sh:1]: `# !/bin/bash` ->  `#!/bin/bash`  remove the blank

File: addons/oceanbase-ce/dataprotection/restore.sh [hunk 2]
```diff
@@ -1,112 +1,61 @@
...
-OlD_IFS=$IFS
 
-archiveStatusFile="tenantStatus.dp"
-provider=
-access_key_id=
-secret_access_key=
-region=
-endpoint=
-bucket=
+SUBDOMAIN=${KB_CLUSTER_COMP_NAME}-headless
```
Comment (Y-Rookie) [inline addons/oceanbase-ce/dataprotection/restore.sh:5]: remove the builti-in `KB_CLUSTER_COMP_NAME` env

File: addons/oceanbase-ce/dataprotection/restore.sh [hunk 3]
```diff
@@ -1,112 +1,61 @@
...
-       return
-    fi
-    # TODO: support cos for 4.2.1 version
-    if [[ ${provider} == "Alibaba" ]];then
-       echo "oss-${DP_STORAGE_REGION}.aliyun-inc.com"
-    fi
-}
+echo "Wait for a while before restoring cluster."
+sleep 30
+primary_host="${KB_CLUSTER_COMP_NAME}-0.${SUBDOMAIN}"
```
Comment (Y-Rookie) [inline addons/oceanbase-ce/dataprotection/restore.sh:12]: ditto

File: addons/oceanbase-ce/reloader/utils.sh [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  eval passwd=\$"${passwdVar}"
+  if [ -n "${passwd}" ]; then
+    echo "${passwd}"
+    return
+  else
+    echo "${OB_ROOT_PASSWD}"
+  fi
+}
+
+ROOT_PASSWD=$(getRootPasswd "${KB_COMP_NAME}")
```
Comment (Y-Rookie) [inline addons/oceanbase-ce/reloader/utils.sh:23]: ditto

File: addons/oceanbase-ce/scripts/utils.sh [hunk 5]
```diff
@@ -1,7 +1,106 @@
...
+  done
+
+  if [ $elapsed_time -ge $wait_time ]; then
+    echo "Failed to init server exit..."
+    exit 1
+  fi
+}
+
 function get_pod_ip {
+  SUBDOMAIN=${KB_CLUSTER_COMP_NAME}-headless
```
Comment (Y-Rookie) [inline addons/oceanbase-ce/scripts/utils.sh:103]: ditto

File: addons/oceanbase-ce/scripts/utils.sh [hunk 6]
```diff
@@ -26,6 +121,49 @@ function get_pod_ip_by_hostname {
...
+    echo "customized_config: $customized_config"
+    default_configs=$customized_config
+  fi
+
+  # get IP address of the current pod
+  curr_pod_ip=$(get_pod_ip ${POD_NAME})
+
+  cluster_id=${OB_CLUSTER_ID:-1}
+
+  /home/admin/oceanbase/bin/observer --appname ${KB_CLUSTER_COMP_NAME} \
```
Comment (Y-Rookie) [inline addons/oceanbase-ce/scripts/utils.sh:146]: ditto

File: addons/oceanbase-ce/templates/_helpers.tpl [hunk 7]
```diff
@@ -73,13 +62,160 @@ Generate scripts configmap
...
-{{- define "oceanbase-release.name" }}
-{{- print "ob-ce" }}
-{{- end }}
\ No newline at end of file
+
+{{- define "oceanbase-ce.compDefName" -}}
+oceanbase-ce-{{ .Chart.Version }}
+{{- end -}}
+
+{{- define "oceanbase-ce.componentDefNamePrefix" -}}
```
Comment (Y-Rookie) [inline addons/oceanbase-ce/templates/_helpers.tpl:77]: use regexp pattern `oceanbase-ce-` would be better, follow the API suggestion, check them all

File: addons/oceanbase-ce/templates/componentdefinition.yaml [hunk 8]
```diff
@@ -1,215 +1,177 @@
...
+          - name: oceanbase-config
+            mountPath: /kb-config
+          - name: oceanbase-sysvars
+            mountPath: /opt/oceanbase/sysvars
+        workingDir: /home/admin/workdir
+        env:
+          - name: LD_LIBRARY_PATH
+            value: /home/admin/oceanbase/lib
+          - name: CLUSTER_NAME
+            value: "$(KB_CLUSTER_COMP_NAME)"
```
Comment (Y-Rookie) [inline addons/oceanbase-ce/templates/componentdefinition.yaml:92]: ditto

File: addons/oceanbase-ce/templates/componentdefinition.yaml [hunk 9]
```diff
@@ -1,215 +1,177 @@
...
-            - --config-path
-            - /tools/config/dbctl/components
-            - oceanbase
-            - getrole
-    labels: &oblabels
-      kubeblocks.io/ready-without-primary: "true"
-    runtime: *obruntime
+  name: {{ include "oceanbase-ce.compDefName" . }}
+  labels:
+    {{- include "oceanbase-ce.labels" . | nindent 4 }}
```
Comment (Y-Rookie) [inline addons/oceanbase-ce/templates/componentdefinition.yaml:7]: add the annotations for helm resource policy keep

Comment (shanshanying) [reply addons/oceanbase-ce/templates/componentdefinition.yaml:7]: Fixed.

File: addons/oceanbase-ce/templates/componentdefinition.yaml (hunk 8 above)
Comment (shanshanying) [reply addons/oceanbase-ce/templates/componentdefinition.yaml:92]: fixed

File: addons/oceanbase-ce/templates/_helpers.tpl (hunk 7 above)
Comment (shanshanying) [reply addons/oceanbase-ce/templates/_helpers.tpl:77]: fixed

File: addons/oceanbase-ce/scripts/utils.sh (hunk 6 above)
Comment (shanshanying) [reply addons/oceanbase-ce/scripts/utils.sh:146]: fixed

File: addons/oceanbase-ce/scripts/utils.sh (hunk 5 above)
Comment (shanshanying) [reply addons/oceanbase-ce/scripts/utils.sh:103]: fixed

File: addons/oceanbase-ce/reloader/utils.sh (hunk 4 above)
Comment (shanshanying) [reply addons/oceanbase-ce/reloader/utils.sh:23]: fixed

File: addons/oceanbase-ce/dataprotection/restore.sh (hunk 3 above)
Comment (shanshanying) [reply addons/oceanbase-ce/dataprotection/restore.sh:12]: fixed

File: addons/oceanbase-ce/dataprotection/restore.sh (hunk 2 above)
Comment (shanshanying) [reply addons/oceanbase-ce/dataprotection/restore.sh:5]: fixed

File: addons/oceanbase-ce/dataprotection/backup.sh (hunk 1 above)
Comment (shanshanying) [reply addons/oceanbase-ce/dataprotection/backup.sh:1]: fixed

**Review by wangyelei — APPROVED** [review]

**Review by Y-Rookie — APPROVED** [review]

**Review by wangyelei — APPROVED** [review]

File: addons-cluster/oceanbase-ce/values.schema.json [hunk 10]
```diff
@@ -4,22 +4,12 @@
...
-                "integer"
-            ],
-            "default": 2,
-            "minimum": 1,
-            "maximum": 2
+            "default": "4.3.0"
         },
         "zoneCount": {
-            "title": "ZoneCount",
+            "title": "ZontCount",
```
Comment (zjx20) [inline addons-cluster/oceanbase-ce/values.schema.json:12]: typo

**Pull request merged**